### DISCUSSION TITLE
How to use a MessagePump?

### DISCUSSION THREAD

**Discussion by hydrargyrum** [Textualize/textual#3835] [Q&A]
I'm developing an app with a `DataTable` and would like to add a `Model` class doing some business logic. This `Model` class should post events that would be handled by the `DataTable` view.
My `Model` is a direct subclass of `MessagePump` but it seems none of the posted messages are received by the view. The `Model` `is_attached` is `False` and so is `is_running`. What could be wrong?

**Comment by davep** [comment]
Something worth mentioning right off I guess, but you wouldn't ordinarily inherit from `MessagePump`; if you want something that's going to be a source of messages that other widgets within the DOM can catch, you'd inherit from at least `Widget`; or you could have other code posting messages at the target widget.

With that out of the way, and if that doesn't clarify things already, perhaps you could provide a [MRE](https://github.com/Textualize/textual/discussions/3361) of wha you're attempting to do.

It might also be worth reviewing [the guide to events and messages](https://textual.textualize.io/guide/events/).
